pyubee/__init__.py

```diff
@@ -136,6 +141,32 @@ def headers(self):
 
         return headers
 
+class DigestAuthAuthenticator(Authenticator):
```

Comment:
This does not seem to do anything by itself, except holding a username/password. Later on, the flow is changed solely for this `Authenticator`. I am for using as much as possible from other libraries, but I am also for consistency in the software and flow. I see you "forgot" to change the `_post()` method (probably don't need it in your case.) When somebody else wants to add support for their router, s/he will have to figure out what is happening/why it is done this way.

One option is to make our own `DigestAuthenticator`, but with its own logic and not relying on `requests`. Sure, this defeats the re-use of the `requests`' `HTTPDigestAuth`, but simplifies the the software in other parts.

Or maybe the current Authenticators can be changed such that these can be passed to `requests.get(auth=..)`/`requests.post(auth=..)`

Or perhaps there is another option, one I haven't thought of.

Reply:
You are right, I didn't need the POST method so I left it untouched. I can change that for sure.

I also would like the code flow to be consistent. However, reinventing the wheel here would just feel stupider. According to all the debug prints in the rest of the `_get()` method it seems like it is rather cumbersome. On top of that, I don't really see the point in not relying in `requests`; it is used later on in both `_get()` and `_post()` anyway. If I wrote the code from scratch, I would've maybe 
1. used the ready Authenticator classes from `requests` instead of creating our own (and if our own is needed, create a child class of an `requests` authenticator)
2. saved that class as an object in the `Ubee` class, so that it can be dynamically used as for example inside the `_get()` as follows: `response = requests.get(url, timeout=HTTP_REQUEST_TIMEOUT, auth=_self.authenticator(self.username, self.password))` where `self.authenticator` was a (derivative of) `requests Authenticator` class.

So yes, I would've preferred the Authenticators of `pyubee` to be compatible with requests. It would remove the reinvention of the wheel I was talking before. After the change, the whole GET or POST methods would be one-liners :) 

I will take a closer look at the comments tomorrow. Thanks for the review. This is just my opinion on this one comment!

Reply:
You are entirely right that it is better to use the tools `requests` provides, thank you for showing this. I guess this was added at some point and others just continued building on the things that were already there.

Though some `Authenticator`s do sneak in their own `GET`s/`POST`s, necessary for some models. Not sure how to handle this, but probably possible.